src/Database/PostgreSQL/Transaction.hs

```diff
@@ -4,6 +4,7 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE ImplicitPrelude            #-}
 {-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE UndecidableInstances       #-}
```

Comment:
UndecidableInstances are par for the course for the mtl instances. Look at the source of mtl for an explanation. 

https://hackage.haskell.org/package/mtl-2.2.1/docs/src/Control-Monad-Reader-Class.html

There isn't much madness to UndecideableInstances, perhaps you mean IncoherentInstances? 

UndecideableInstances don't lead to unpredictable instances, the compiler might hit a recursion limit, but I don't think that is known to happen with the way people use mtl.